qiskit_nature/second_q/operators/polynomial_tensor.py

```diff
@@ -105,58 +246,97 @@ def is_empty(self) -> bool:
         """Returns whether this tensor is empty or not."""
         return len(self) == 0
 
-    def __getitem__(self, __k: str) -> (np.ndarray | Number):
-        """
-        Returns value matrix in the `PolynomialTensor`.
+    @_optionals.HAS_SPARSE.require_in_call
+    def contains_sparse(self) -> bool:
+        """Returns whether this tensor contains any sparse matrix."""
+        return any(isinstance(self[key], SparseArray) for key in self)
+
+    def __getitem__(self, __k: str) -> (np.ndarray | SparseArray | Number):
+        """Gets the value from the ``PolynomialTensor``.
 
         Args:
             __k: operator key string in the `PolynomialTensor`.
+
         Returns:
-            Value matrix corresponding to the operator key `__k`
+            Value corresponding to the operator key `__k`.
         """
         return self._data.__getitem__(__k)
 
     def __len__(self) -> int:
-        """
-        Returns length of `PolynomialTensor`.
-        """
+        """Returns the length of the ``PolynomialTensor``."""
         return self._data.__len__()
 
     def __iter__(self) -> Iterator[str]:
-        """
-        Returns iterator of the `PolynomialTensor`.
-        """
+        """Returns an iterator of the ``PolynomialTensor``."""
         return self._data.__iter__()
 
+    @_optionals.HAS_SPARSE.require_in_call
+    def todense(self) -> PolynomialTensor:
+        """Converts all internal matrices to dense numpy arrays."""
```

Comment:
Could we say for this and `tosparse` below more like "Returns a new instance where all matrices are now dense numpy array" to make it clear its not an inplace conversion.